rewrite-java/src/main/java/org/openrewrite/java/ChangeTypeInStringLiteral.java

```diff
@@ -0,0 +1,118 @@
+        // is a temporary workaround until releases of workers/CLI that include the defensive
+        // coding in Recipe.
+        //noinspection ConstantValue
+        if (oldFullyQualifiedTypeName == null || newFullyQualifiedTypeName == null) {
+            return getDisplayName();
+        }
+
+        String oldShort = oldFullyQualifiedTypeName.substring(oldFullyQualifiedTypeName.lastIndexOf('.') + 1);
+        String newShort = newFullyQualifiedTypeName.substring(newFullyQualifiedTypeName.lastIndexOf('.') + 1);
+        if (oldShort.equals(newShort)) {
+            return String.format("`%s` to `%s`",
+                    oldFullyQualifiedTypeName,
+                    newFullyQualifiedTypeName);
+        } else {
+            return String.format("`%s` to `%s`",
+                    oldShort, newShort);
+        }
+    }
+
+    @Override
+    public String getDescription() {
+        return "Change a given type to another when used in a String literal.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public boolean isAcceptable(SourceFile sourceFile, ExecutionContext ctx) {
+                return sourceFile instanceof JavaSourceFile;
+            }
+
+            @Override
+            public @Nullable Tree preVisit(@Nullable Tree tree, ExecutionContext ctx) {
+                stopAfterPreVisit();
+                if (tree instanceof J) {
+                    return new JavaChangeTypeLiteralVisitor(oldFullyQualifiedTypeName, newFullyQualifiedTypeName).visit(tree, ctx, requireNonNull(getCursor().getParent()));
+                }
+                return tree;
+            }
+        };
+    }
+
+    private static class JavaChangeTypeLiteralVisitor extends JavaVisitor<ExecutionContext> {
+        private final String newFullyQualifiedTypeName;
+        private final Pattern stringLiteralPattern;
+
+        private JavaChangeTypeLiteralVisitor(String oldFullyQualifiedTypeName, String newFullyQualifiedTypeName) {
+            this.newFullyQualifiedTypeName = newFullyQualifiedTypeName;
+            this.stringLiteralPattern = Pattern.compile("\\b" + oldFullyQualifiedTypeName + "\\b");
+        }
+
+        @Override
+        public J visitLiteral(J.Literal literal, ExecutionContext ctx) {
+            J.Literal lit = literal;
+            if (literal.getType() == JavaType.Primitive.String && lit.getValue() != null) {
+                Matcher matcher = stringLiteralPattern.matcher((String) lit.getValue());
+                if (matcher.find()) {
+                    lit = lit.withValue(matcher.replaceAll(newFullyQualifiedTypeName))
+                            .withValueSource(stringLiteralPattern.matcher(lit.getValueSource()).replaceAll(newFullyQualifiedTypeName));
+                }
+            }
+            return super.visitLiteral(lit, ctx);
+        }
+    }
```

Comment:
Why these levels of indirection? `JavaVisitor` already has a `isAcceptable` that checks `sourceFile instanceof JavaSourceFile`.
```suggestion
    public TreeVisitor<?, ExecutionContext> getVisitor() {
        final Pattern stringLiteralPattern = Pattern.compile("\\b" + oldFullyQualifiedTypeName + "\\b");
        return new JavaVisitor<ExecutionContext>() {
            @Override
            public J visitLiteral(J.Literal literal, ExecutionContext ctx) {
                J.Literal lit = literal;
                if (literal.getType() == JavaType.Primitive.String && lit.getValue() != null) {
                    Matcher matcher = stringLiteralPattern.matcher((String) lit.getValue());
                    if (matcher.find()) {
                        lit = lit.withValue(matcher.replaceAll(newFullyQualifiedTypeName))
                                .withValueSource(stringLiteralPattern.matcher(lit.getValueSource()).replaceAll(newFullyQualifiedTypeName));
                    }
                }
                return super.visitLiteral(lit, ctx);
            }
        };
    }
```